opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/PushDownQueryBuilder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.scan;
+
+import org.opensearch.sql.opensearch.request.OpenSearchRequestBuilder;
+import org.opensearch.sql.planner.logical.LogicalFilter;
+import org.opensearch.sql.planner.logical.LogicalHighlight;
+import org.opensearch.sql.planner.logical.LogicalLimit;
+import org.opensearch.sql.planner.logical.LogicalNested;
+import org.opensearch.sql.planner.logical.LogicalPaginate;
+import org.opensearch.sql.planner.logical.LogicalProject;
+import org.opensearch.sql.planner.logical.LogicalSort;
+
+/**
+ * Translates a logical query plan into OpenSearch DSL and an appropriate request.
+ */
+public interface PushDownQueryBuilder {
```

Comment:
Could you elaborate the relation between this and `OpenSearchRequestBuilder`?
So this builder builds another builder `OpenSearchRequestBuilder` and meanwhile returns the subclass rather than its interface `ExecutableRequestBuilder`?

Reply:
This changed and probably warrants another look.

`PushDownQueryBuilder` is interface a class that can implement push down rules needs to support. 
It's implemented by `OpenSearchIndexScanQueryBuilder` and `OpenSearchIndexScanAggregationBuilder` because simple and aggregation queries have different pushdown logic for limit and sort.

For each push down operation,`OpenSearchRequestBuilder` delegates the actual implementation to either `OpenSearchIndexScanQueryBuilder`. One of those implementations swaps the delegate from `OpenSearchIndexScanQueryBuilder` to `OpenSearchIndexScanAggregationBuilder`.

Reply:
Maybe post this as a code comment or in the doc?